src/FilterButtonList.js

```diff
@@ -1,40 +1,51 @@
 import F from 'futil'
 import _ from 'lodash/fp'
 import React from 'react'
+import { observer } from 'mobx-react'
 import { setDisplayName } from 'recompose'
 import { Dynamic, Flex } from './greyVest'
-import { CheckButton } from './purgatory'
-import { withNode, withLoader } from './utils/hoc'
+import { CheckButton, ModalPicker } from './purgatory'
+import { withNode } from './utils/hoc'
 import { withTheme } from './utils/theme'
 import styles from './styles'
+import { newNodeFromField } from './utils/search'
+import { fieldsToOptions, getGroupFields } from './FilterAdder'
 
 let FilterButtonItem = _.flow(
   setDisplayName('FilterButtonItem'),
-  withLoader,
   withTheme
 )(
   ({
     node,
     tree,
     fields,
     mapNodeToProps,
-    theme: { Button, UnmappedNodeComponent, Modal },
+    theme: { Button, FilterButton = Button, UnmappedNodeComponent, Modal },
   }) => {
     let mappedProps = mapNodeToProps(node, fields)
     let modal = React.useState(false)
     let title = // we really need a title, so here's every possible fallback
       _.get('label', mappedProps) ||
+      _.get([node.key, 'label'], fields) ||
       _.get([node.field, 'label'], fields) ||
       node.field ||
       node.key
     let description = _.get('description', mappedProps)
     return (
-      <>
-        <CheckButton checked={node.hasValue} onClick={F.on(modal)}>
+      <div>
```

Comment:
Why change from the fragment here?

Reply:
We need the buttons to be in containers so they are aligned properly with the parent's flex